### DISCUSSION TITLE
How to merge author's changes into my forked repo?

### DISCUSSION THREAD

**Discussion by robertandrews** [community/community#23484] [New to GitHub]
<p>I took a <a href="https://github.com/tidythemes/blankslate" rel="noopener nofollow ugc">WordPress theme</a> from GitHub and <a href="https://github.com/robertandrews/blankslate" rel="noopener nofollow ugc">forked it</a>. Now I want to incorporate some of the changes the original author has made since I did this. How can I do this in VS Code?</p>
<p>The only way I have seen is in <a href="https://github.com" rel="noopener nofollow ugc">github.com</a> itself, on my forked version: “Fetch upstream”, which would open a pull request to myself, which can be seen either in VS Code or <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>… except, it’s a request I’m apparently not allowed to self-approve.</p>
<p>Happy to learn the core processes at work (ie. is it something to do with <code>fetch</code>/<code>checkout</code>/<code>rebase</code>?), but would like to be doing it in the VS Code interface, not comment line.</p>
<p>I have git, remotes/Source control and the GitHub extension set up in my VS Code.</p>

**Comment by robertandrews** [comment]
<p>Here’s something I just tried in VS Code…</p>
<ul>
<li>“Source Control” panel &gt; “…” menu item &gt; “Remote” &gt; “Add remote…” – added the original author’s tidythemes/blankslate as a remote.</li>
<li>“Source Control” panel &gt; “…” menu item &gt; "Fetch – this did nothing.</li>
<li>“Source Control” panel &gt; “…” menu item &gt; “Pull, Push” &gt; “Pull from…” &gt; Choose original from list of remotes…</li>
</ul>
<p>This pulled in a host of conflicts etc to attend to, which I could merge using options displayed in-line in code.</p>
<p>Did I get it right? I assumed fetch/checkout/rebase or similar were the thing.</p>
<p>One thing I noticed is that the original repo was already listed in the remotes picker for “Pull from…” - I guess I didn’t need to add a new remote?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="robertandrews" data-post="2" data-topic="251075">
<div class="title">
<div class="quote-controls"></div>
 robertandrews:</div>
<blockquote>
<p>“Source Control” panel &gt; “…” menu item &gt; "Fetch – this did nothing.</p>
</blockquote>
</aside>
<p>Fetch only updates the information Git has about the remote repository. It only retrieves data, it does not combine them with your local data in any way.</p>
<p>What you need to include the upstream changes in your own branch is either merge or rebase, which of them depends on how you want to manage your branches.</p>
<p>Pull is a combined fetch &amp; merge operation.</p>

**Comment by robertandrews** [comment]
<p>How should this be done exactly… ?</p>
<p><strong>Option: Via the GitHub web UI</strong>, <a href="https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork" rel="noopener nofollow ugc">instructions suggest</a>:</p>
<ul>
<li>Click “Fetch upstream”</li>
<li>Click “Fetch and merge” button</li>
<li>“If the changes from the upstream repository cause conflicts, GitHub will prompt you to create a pull request to resolve the conflicts.” a) Does this mean a prompt to approve every variance? b) Do these resolution prompts appear on the VS Code side, too? Maybe only after pressing the Sync icon or similar?</li>
</ul>
<p><strong>Option: More importantly, how to do this all inside VS Code</strong>?</p>
<p>I think it needs to start with me properly understanding “<strong>remotes</strong>”. It all depends on you having the upstream added as a remote first, right? Circumstances where this may not initially be the case - when I have fetched down my fork from GitHub (maybe this would make <em>my</em> GH repo the remote?)?..</p>
<p>And then is it just a matter of choosing <strong>Merge</strong> from the remote? Are there no changes/variances indicators until you do that?</p>
<p>Does none of this require a Pull Request?</p>
<p>Thanks.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Note: I don’t use VS code, what I’m describing are the general Git terms. You’ll have to figure out how (or if) they’re available in a GUI, or maybe someone else can expand on that.</p>
<aside class="quote no-group" data-username="robertandrews" data-post="4" data-topic="251075">
<div class="title">
<div class="quote-controls"></div>
 robertandrews:</div>
<blockquote>
<p>I think it needs to start with me properly understanding “ <strong>remotes</strong> ”. It all depends on you having the upstream added as a remote first, right? Circumstances where this may not initially be the case - when I have fetched down my fork from GitHub (maybe this would make <em>my</em> GH repo the remote?)?..</p>
</blockquote>
</aside>
<p>When you clone a repository (e.g. your fork on GitHub), that repository becomes by default a remote called <code>origin</code> in the new local repository. It’s important to understand that your local repository is an equally full, independent repository (unless you take specific steps to avoid that, like using a shallow clone). In your local repository you can add, remove, and rename remotes as you like.</p>
<p>See: <a href="https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes" class="inline-onebox">Git - Working with Remotes</a></p>
<aside class="quote no-group" data-username="robertandrews" data-post="4" data-topic="251075">
<div class="title">
<div class="quote-controls"></div>
 robertandrews:</div>
<blockquote>
<p>And then is it just a matter of choosing <strong>Merge</strong> from the remote? Are there no changes/variances indicators until you do that?</p>
</blockquote>
</aside>
<p>Your local repository becomes aware of changes in a remote once you fetch that remote. Fetch updates the data you have locally about the state of the remote: Branches, commits, etc. It does not modify your own working tree or branches. At that point you can compare (<code>git diff</code>), merge, or check out branches.</p>

**Comment by robertandrews** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> Thanks for the lessons here.</p>
